### PR TITLE
Email auth factor

### PR DESCRIPTION
**Pull request by pfrazee**
- [x] Add a setting for enabling email 2fa
- [x] Add prompt for 2fa code in login
- [x] Wire up to backend

## To test:

- Create a test account.
- Go to settings. Enable 2FA. This will trigger email verification. Complete verification and it will be enabled.
- Log out. Log back in. You will be prompted to provide the code.
- Go to settings. Disable 2FA. This will email a confirmation code. Enter the code and it will be disabled.
- Log out. Log back in. Code no longer required.

## Video

https://github.com/bluesky-social/social-app/assets/1270099/6129f917-329f-48d1-8829-c74c81e93e88

### CODE REVIEW

File: src/view/screens/Settings/DisableEmail2FADialog.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            style={[a.text_2xl, a.font_bold, t.atoms.text]}>
+            <Trans>Disable Email 2FA</Trans>
+          </Text>
+          <P
+            nativeID="dialog-description"
+            style={[a.text_sm, t.atoms.text, a.leading_snug]}>
+            {stage === Stages.ConfirmCode ? (
+              <Trans>
+                An email has been sent to{' '}
+                {currentAccount?.email || '(no email)'}. It includes a
```
Comment (gaearon) [inline src/view/screens/Settings/DisableEmail2FADialog.tsx:93]: Is this some sort of a corner case?

File: src/screens/Login/LoginForm.tsx [hunk 2]
```diff
@@ -215,6 +227,36 @@ export const LoginForm = ({
...
         </View>
       </View>
+      {isAuthFactorTokenNeeded && (
+        <View>
+          <TextField.LabelText>
+            <Trans>2FA Confirmation</Trans>
+          </TextField.LabelText>
+          <TextField.Root>
+            <TextField.Icon icon={Ticket} />
+            <TextField.Input
```
Comment (gaearon) [inline src/screens/Login/LoginForm.tsx:237]: Pressing Enter while in this field doesn't submit the form which feels annoying

File: src/view/screens/Settings/DisableEmail2FADialog.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+              </Trans>
+            ) : (
+              <Trans>
+                To disable the email 2FA method, please verify your access to
+                the email address.
+              </Trans>
+            )}
+          </P>
+
+          {error ? <ErrorMessage message={error} /> : undefined}
```
Comment (gaearon) [inline src/view/screens/Settings/DisableEmail2FADialog.tsx:104]: Screenshot 2024-04-19 at 04 22 35

Can maybe make this message less technical? Not obvious if 2FA error or the app bugged.

File: src/screens/Login/LoginForm.tsx [hunk 4]
```diff
@@ -100,14 +107,19 @@ export const LoginForm = ({
           service: serviceUrl,
           identifier: fullIdent,
           password,
+          authFactorToken,
```
Comment (gaearon) [inline src/screens/Login/LoginForm.tsx:110]: Maybe trim spaces? For easier copy paste.

**Review by gaearon — APPROVED** [review]
lg, seems to work!

File: src/view/screens/Settings/DisableEmail2FADialog.tsx (hunk 1 above)
Comment (pfrazee) [reply src/view/screens/Settings/DisableEmail2FADialog.tsx:93]: Yeah that really shouldnt be possible

File: src/view/screens/Settings/DisableEmail2FADialog.tsx (hunk 3 above)
Comment (pfrazee) [reply src/view/screens/Settings/DisableEmail2FADialog.tsx:104]: Good catch, fixed

File: src/screens/Login/LoginForm.tsx (hunk 4 above)
Comment (pfrazee) [reply src/screens/Login/LoginForm.tsx:110]: Good catch, done

File: src/screens/Login/LoginForm.tsx (hunk 2 above)
Comment (pfrazee) [reply src/screens/Login/LoginForm.tsx:242]: Fixed

**Pull request merged**